router_client.go

```diff
@@ -32,6 +34,14 @@ type RouterClient interface {
 	// payment update stream and an error stream.
 	TrackPayment(ctx context.Context, hash lntypes.Hash) (
 		chan PaymentStatus, chan error, error)
+
+	// SubscribeHtlcEvents subscribes to a stream of htlc events from the
+	// router. In the absence of an shared interface in htlcswitch/
+	// htlcnotfier, we use an empty interface for htlc events. Note that
```

Comment:
hm, would it be better to create such a shared interface/struct for events? Seems like subscribers to this RPC shouldn't have to deal with the internal lnd types.

Reply:
Kind of an open question for this repo, whether to go back to lnd's types or create entirely new ones. We are already using quite a few internal types (contract state for invoices, some payment structs), but have also created new ones. I went with the lnd types here because there's so much that would need to be duplicated, 4x event types + 22 different enum values, and then we need to keep track of three things, rather than just lnrpc+lnd.

The empty interface isn't nice though, WDYT about wrapping the lnd types in a set of structs with a common interface for now, and eventually adding the interface in lnd itself?
```
type ForwardingEvent struct{
  htlcSwitch.Forwardingevent
}
``

Reply:
Discussed offline, just going with protos for now.